src/main/java/com/CezaryZal/manager/filters/comparator/PasswordComparator.java

```diff
@@ -1,13 +1,14 @@
 package com.CezaryZal.manager.filters.comparator;
 
 import com.CezaryZal.exceptions.IncorrectLoginOrPasswordSecurityException;
+import org.springframework.security.crypto.bcrypt.BCrypt;
 import org.springframework.stereotype.Service;
 
 @Service
 public class PasswordComparator {
 
-    public void throwIsNotEqualsPassword(String passwordFromUserLogin, String userPasswordFromDb) {
-        if (!passwordFromUserLogin.equals(userPasswordFromDb)){
+    public void throwIfIsNotEqualsPassword(String passwordFromUserLogin, String userPasswordFromDb) {
+        if (!BCrypt.checkpw(passwordFromUserLogin, userPasswordFromDb)){
             throw new IncorrectLoginOrPasswordSecurityException("Login or password incorrectly entered");
```

Comment:
Logowanie ma kilka etapów.
Przychodzi request z loginem i hasłem.
1. Sprawdzasz czy istnieje login. Jeżeli nie istnieje login to rzucasz exception AccountNotFound
2.Sprawdzasz czy hasło jest dobre. Jeżeli nie to rzucasz exception że InvalidPasswordException
3. Jeżeli któryś z tych exceptionów poleciał, to wyżej (w miejscu validacji czy tego usera możesz zalogować łapiesz go i rzucasz InvalidLoginOrPassword

Reply:
Ok, myślałem że skoro i tak na końcu dajemy ogólny wyjątek to nie ma sensu tego rozbijać, ale to przydatne przy czytaniu logów.

Reply:
Ok, obsługę pierwszych wyjątków zrobiłem przez @ControllerAdvice w RestResponseEntityExceptionHandler przez wyłapanie i zmianę Stringa w body.
W taki sposób czy chodziło Ci o Try()catch()?

Reply:
Tak, przez try&catch.
Jeżeli złapiesz AccountNotFount i InvalicPassword i w ich miejsce rzucisz InvalidUserNameOrPassword to nie musisz mieć handlerów dla AccountNotFoun i Invalid Password